packages/medusa/src/services/draft-order.ts

```diff
@@ -0,0 +1,437 @@
+  static readonly Events = {
+    CREATED: "draft_order.created",
+    UPDATED: "draft_order.updated",
+  }
+
+  protected readonly manager_: EntityManager
+  protected readonly draftOrderRepository_: typeof DraftOrderRepository
+  protected readonly paymentRepository_: typeof PaymentRepository
+  protected readonly orderRepository_: typeof OrderRepository
+  protected readonly eventBus_: EventBusService
+  protected readonly cartService_: CartService
+  protected readonly lineItemService_: LineItemService
+  protected readonly productVariantService_: ProductVariantService
+  protected readonly shippingOptionService_: ShippingOptionService
+
+  constructor({
+    manager,
+    draftOrderRepository,
+    paymentRepository,
+    orderRepository,
+    eventBusService,
+    cartService,
+    lineItemService,
+    productVariantService,
+    shippingOptionService,
+  }: InjectedDependencies) {
+    super()
+
+    this.manager_ = manager
+    this.draftOrderRepository_ = draftOrderRepository
+    this.paymentRepository_ = paymentRepository
+    this.orderRepository_ = orderRepository
+    this.lineItemService_ = lineItemService
+    this.cartService_ = cartService
+    this.productVariantService_ = productVariantService
+    this.shippingOptionService_ = shippingOptionService
+    this.eventBus_ = eventBusService
+  }
+
+  withTransaction(transactionManager: EntityManager): DraftOrderService {
+    if (!transactionManager) {
+      return this
+    }
+
+    const cloned = new DraftOrderService({
+      manager: transactionManager,
+      draftOrderRepository: this.draftOrderRepository_,
+      paymentRepository: this.paymentRepository_,
+      orderRepository: this.orderRepository_,
+      lineItemService: this.lineItemService_,
+      cartService: this.cartService_,
+      productVariantService: this.productVariantService_,
+      shippingOptionService: this.shippingOptionService_,
+      eventBusService: this.eventBus_,
+    })
+
+    cloned.transactionManager_ = transactionManager
+
+    return cloned
+  }
+
+  /**
+   * Retrieves a draft order with the given id.
+   * @param {string} id - id of the draft order to retrieve
+   * @param {object} config - query object for findOne
+   * @return {Promise<DraftOrder>} the draft order
+   */
+  async retrieve(id: string, config = {}): Promise<DraftOrder> {
+    return await this.atomicPhase_(
+      async (transactionManager: EntityManager) => {
+        const draftOrderRepo = transactionManager.getCustomRepository(
+          this.draftOrderRepository_
+        )
+
+        const validatedId = this.validateId_(id)
+
+        const query = this.buildQuery_({ id: validatedId }, config)
+        return await draftOrderRepo.findOne(query).then((draftOrder) => {
+          if (draftOrder) {
+            return draftOrder
+          }
+
+          throw new MedusaError(
+            MedusaError.Types.NOT_FOUND,
+            `Draft order with ${id} was not found`
+          )
+        })
```

Comment:
suggestion: Use the general convention of awaiting instead of `.then` chaining. 

See [CustomerService for reference](https://github.com/medusajs/medusa/blob/7a1e394b9df9aaf95bcc885685f99ede1dd7aaa9/packages/medusa/src/services/customer.js#L243).